rust/hyperlane-core/src/error.rs

```diff
@@ -134,6 +134,10 @@ pub enum ChainCommunicationError {
     /// Rpc client error
     #[error(transparent)]
     RpcClientError(#[from] RpcClientError),
+    /// Tokio join error
+    #[cfg(feature = "fallback-provider")]
```

Comment:
kinda awkward to have this feature relate to something that's used outside of fallback provider. maybe a case for a rename of the feature? don't feel strongly

Reply:
renamed the feature to `async`